packages/sortable/src/components/SortableContext.tsx

```diff
@@ -60,6 +60,21 @@ export function SortableContext({
       ),
     [userDefinedItems]
   );
+  // add a runtime type check for non-TS users to avoid confusion around
+  // non-string IDs. See the following links for context on why this is useful.
+  // https://github.com/clauderic/dnd-kit/issues/604
+  // https://github.com/clauderic/dnd-kit/discussions/357#discussioncomment-1018851
+  // https://github.com/clauderic/dnd-kit/issues/606
+  //
+  // the process.env guard will cause this check to be removed from the bundle
+  // at build time by most modern build tools
+  if (process.env.NODE_ENV !== 'production') {
+    if (items.every((i) => typeof id === 'string')) {
```

Comment:
Because SortableContext is relatively high up in the render tree, I don't _think_ this will cause perf issues in development but because it's iterating over the items array, it does have the potential to.

This check could be memoized but it would need to use something besides React.useMemo in order to avoid the hooks linter complaining about hooks inside a conditional (even if that conditional will never change at runtime...)